docs/advanced/integration_tests.md

```diff
@@ -0,0 +1,130 @@
+
+If you have a Gosoline application that takes its input from a stream, you need a test which can run your application locally, send data to a stream, and check your application's output for correctness. 
+
+The `stream-consumer` application, found in `advanced/stream-consumer`, reads unsigned integers from an `inputEvent` sqs queue, increments them by one and publishes them to the `publisher-outputEvent` SNS topic. Both the SQS queue which provides it with input, and the SNS topic to which it publishes its results, are configured in its `config.dist.yml` file. Below is an extract from its integration test:
+
+[embedmd]:# (../../examples/advanced/stream-consumer-test/stream_consumer_test.go /func \(s \*ConsumerTestSuite\) SetupSuite/ /\n}/)
+```go
+func (s *ConsumerTestSuite) SetupSuite() []suite.Option {
+	return []suite.Option{
+		suite.WithLogLevel("debug"),
+		suite.WithConfigFile("../stream-consumer/config.dist.yml"),
+		suite.WithModule("consumer", stream.NewConsumer("consumer", consumer.NewConsumer())),
+	}
+}
+```
+
+It is making use of the same `config.dist.yml` file as `stream-consumer`, and it will be using the module created by _consumer.NewConsumer_. The difference between a _StreamTestCase_ test and an _ApiServerTestCase_ can be seen in its test cases:
+
+[embedmd]:# (../../examples/advanced/stream-consumer-test/stream_consumer_test.go /func \(s \*ConsumerTestSuite\) TestSuccess/ /\n}/)
+```go
+func (s *ConsumerTestSuite) TestSuccess() *suite.StreamTestCase {
+	input := mdl.Uint(5)
+
+	return &suite.StreamTestCase{
+		Input: map[string][]suite.StreamTestCaseInput{
+			"consumer": {
+				{
+					Attributes: nil,
+					Body:       &input,
+				},
+			},
+		},
+		Assert: func() error {
+			result := mdl.Uint(0)
+			s.Env().StreamOutput("publisher-outputEvent").Unmarshal(0, result)
+
+			s.Equal(mdl.Uint(6), result)
+
+			return nil
+		},
+	}
+}
+```
+
+Similarly to an _ApiServerTestCase_, this _StreamTestCase_ is an object defining an input and an `Assert` function. Gosoline will run the module, send this input into its SNS queue, then run `Assert`. The `Assert` function reads the first element from a stream, _publisher-outputEvent_, then compare it with an expected result.
+
+## Shared environment
+
+One of the options for a Gosoline _suite_ integraiton test is `suite.WithSharedEnvironment()`. When this option is off, each test case will run in its own environment. For example, the fixtures are being loaded for every test case, and any change to a database or stream will only last during that test case alone. When the option is one, the environment is created only once and used by all the test cases, and any change done by one test case will be available to the ones who follow.
+
+## Auto detect components
+
+Another option each _suite_ test offers is `WithoutAutoDetectedComponents`. This simply add one extra options to the test, which tells it to skip one of the components configured in any potential `config.dist.yml`.
```

Comment:
One component? How is that chosen? Randomly? And which components? Do we already know what components are at this point?